jlm/hls/backend/rvsdg2rhls/mem-queue.cpp

```diff
@@ -287,10 +284,9 @@ separate_load_edge(
         else
         {
           // end of loop
-          auto load_user_input = dynamic_cast<jlm::rvsdg::simple_input *>(addr_edge_user);
-          JLM_ASSERT(load_user_input);
-          JLM_ASSERT(
-              dynamic_cast<const jlm::hls::branch_op *>(&load_user_input->node()->GetOperation()));
+          auto load_user_input = jlm::util::AssertedCast<jlm::rvsdg::simple_input>(addr_edge_user);
+          jlm::util::AssertedCast<const jlm::hls::branch_op>(
```

Comment:
This here should have stayed a JLM_ASSERT() as the result of the AssertedCast is not used.

Reply:
Reverted